pgmpy/sampling/Sampling.py

```diff
@@ -331,7 +331,7 @@ def _get_kernel_from_markov_model(self, model):
             factor = factors_dict[var]
             scope = set(factor.scope())
             for tup in itertools.product(*[range(card) for card in other_cards]):
-                states = [State(var, s) for var, s in zip(other_vars, tup) if var in scope]
+                states = [State(first_var, s) for first_var, s in zip(other_vars, tup) if first_var in scope]
```

Comment:
@fabriziov Can you change this name to `v`? `first_var` doesn't seem a good name.